src/main/scala/mesosphere/mesos/RunSpecOfferMatcher.scala

```diff
@@ -9,12 +10,14 @@ import org.apache.mesos.Protos.Offer
 import org.slf4j.LoggerFactory
 
 import scala.collection.immutable.Seq
+import scala.concurrent.duration._
 
 object RunSpecOfferMatcher {
 
   val log = LoggerFactory.getLogger(getClass)
 
-  def matchOffer(runSpec: RunSpec, offer: Offer, otherInstances: => Seq[Instance], givenAcceptedResourceRoles: Set[String]): ResourceMatchResponse = {
+  def matchOffer(runSpec: RunSpec, offer: Offer, otherInstances: => Seq[Instance],
+    givenAcceptedResourceRoles: Set[String], drainingTime: FiniteDuration)(implicit clock: Clock): ResourceMatchResponse = {
```

Comment:
We can make this PR much smaller by adding default value for `drainingTime`. On the other hand I prefer to avoid default values and be explicit.